services/ImageService_test.go

```diff
@@ -0,0 +1,1255 @@
+	imageUpdateNoUser       = "records/image/updateNoUser"
+	imageUpdateUnknown      = "records/image/updateUnknown"
+
+	imageChangeType        = "records/image/changeType"
+	imageChangeTypeUnknown = "records/image/changeTypeUnknown"
+	imageChangeTypeNoImage = "records/image/changeTypeNoImage"
+
+	imageChmod              = "records/image/chmod"
+	imagePermRequestDefault = "records/image/chmodPermReqDefault"
+	imageChmodUnknown       = "records/image/chmodUnknown"
+	imageChmodNoImage       = "records/image/chmodNoImage"
+
+	imageChown               = "records/image/chown"
+	imageOwnershipReqDefault = "records/image/chownDefault"
+	imageChownUnknown        = "records/image/chownUnknown"
+	imageChownNoImage        = "records/image/chownNoImage"
+
+	imageRename        = "records/image/rename"
+	imageRenameEmpty   = "records/image/renameEmpty"
+	imageRenameUnknown = "records/image/renameUnknown"
+	imageRenameNoImage = "records/image/renameNoImage"
+
+	imageEnable         = "records/image/enable"
+	imageDisable        = "records/image/disable"
+	imageDisableUnknown = "records/image/disableUnknown"
+	imageDisableNoImage = "records/image/disableNoImage"
+
+	imagePersistent           = "records/image/persistent"
+	imageNonpersistent        = "records/image/nonpersistent"
+	imageNonpersistentUnknown = "records/image/nonpersistentUnknown"
+	imageNonpersistentNoImage = "records/image/nonpersistentNoImage"
+
+	imageRetrieveInfo        = "records/image/retrieveInfo"
+	imageRetrieveInfoUnknown = "records/image/retrieveInfoUnknown"
+
+	imageListAllPrimaryGroup = "records/image/listAllPrimaryGroup"
+	imageListAllUser         = "records/image/listAllUser"
+	imageListAllAll          = "records/image/listAllAll"
+	imageListAllUserGroup    = "records/image/listAllUserGroup"
+
+	imageListAllForUser        = "records/image/listAllForUser"
+	imageListAllForUserUnknown = "records/image/listAllForUserUnknown"
+	imageListAllForUserEmpty   = "records/image/listAllForUserEmpty"
+)
+
+var _ = ginkgo.Describe("Image Service", func() {
+	var (
+		recName string
+		rec     *recorder.Recorder
+		client  *onego.Client
+		err     error
+	)
+
+	ginkgo.JustBeforeEach(func() {
+		// Start recorder
+		rec, err = recorder.New(recName)
+		if err != nil {
+			return
+		}
+
+		rec.SetMatcher(func(r *http.Request, i cassette.Request) bool {
+			var b bytes.Buffer
+			if _, err = b.ReadFrom(r.Body); err != nil {
+				return false
+			}
+			r.Body = ioutil.NopCloser(&b)
+			return cassette.DefaultMatcher(r, i) && (b.String() == "" || b.String() == i.Body)
+		})
+
+		// Create an HTTP client and inject our transport
+		clientHTTP := &http.Client{
+			Transport: rec, // Inject as transport!
+		}
+
+		// create onego client
+		client = onego.CreateClient(endpoint, token, clientHTTP)
+		if client == nil {
+			err = fmt.Errorf("no client")
+			return
+		}
+	})
+
+	ginkgo.AfterEach(func() {
+		rec.Stop()
+	})
+
+	ginkgo.Describe("allocate image", func() {
+		var (
```

Comment:
A lot of variables (not just here) are identical for multiple tests. To avoid code repetition define such variables in main `Describe` block.